examples/stress_tests/xr_cubes_bench.rs

```diff
@@ -0,0 +1,150 @@
+//! A cube-spawning benchmark.
+//!
+//! Press the arrow keys to adjust the number of cubes, then press R to spawn
+//! them.
+
+use bevy::diagnostic::{FrameTimeDiagnosticsPlugin, LogDiagnosticsPlugin};
+use bevy::prelude::*;
+
+#[derive(Component)]
+struct Cube;
+
+#[derive(Resource)]
+struct Spawner {
+    pub width: i32,
+    pub height: i32,
+    pub cube_size: f32,
+    pub should_spawn: bool,
+    pub material: Option<Handle<StandardMaterial>>,
+    pub mesh: Option<Handle<Mesh>>,
+}
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_plugins(FrameTimeDiagnosticsPlugin)
+        .add_plugins(LogDiagnosticsPlugin::default())
+        .add_systems(Startup, setup)
+        .insert_resource(Spawner {
+            should_spawn: true,
+            width: 20,
+            height: 100,
+            cube_size: 0.01,
+            mesh: None,
+            material: None,
+        })
+        .add_systems(Update, (modify_spawn_parameters, spawn_boxes_system))
+        .run();
+}
+
+/// set up a simple 3D scene
+fn setup(
+    mut commands: Commands,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut materials: ResMut<Assets<StandardMaterial>>,
+) {
+    // plane
+    commands.spawn(PbrBundle {
+        mesh: meshes.add(shape::Plane::from_size(5.0)),
+        material: materials.add(Color::rgb(0.3, 0.5, 0.3)),
+        ..default()
+    });
+    // cube
+    commands.spawn(PbrBundle {
+        mesh: meshes.add(Mesh::from(shape::Cube { size: 0.1 })),
+        material: materials.add(Color::rgb(0.8, 0.7, 0.6)),
+        transform: Transform::from_xyz(0.0, 0.5, 0.0),
+        ..default()
+    });
+    // cube
+    commands.spawn(PbrBundle {
+        mesh: meshes.add(Mesh::from(shape::Cube { size: 0.1 })),
+        material: materials.add(Color::rgb(0.8, 0.0, 0.0)),
+        transform: Transform::from_xyz(0.0, 0.5, 1.0),
+        ..default()
+    });
+    // camera
+    commands.spawn(Camera3dBundle {
+        transform: Transform::from_xyz(-2.0, 2.5, 5.0).looking_at(Vec3::ZERO, Vec3::Y),
+        ..default()
+    });
+}
+
+fn modify_spawn_parameters(
+    mut commands: Commands,
+    things_query: Query<Entity, With<Cube>>,
+    keys: Res<ButtonInput<KeyCode>>,
+    mut spawner: ResMut<Spawner>,
+) {
+    if keys.just_released(KeyCode::ArrowUp) {
+        spawner.height += 10;
+        println!("Boxes: {}", spawner.width * spawner.height);
```

Comment:
you should not use `println` in examples, even more so in benchmarks as it can block.